plugins/toolbox-search/src/block_searcher.ts

```diff
@@ -59,10 +74,10 @@ export class BlockSearcher {
    * Filters the available blocks based on the current query string.
    *
    * @param query The text to use to match blocks against.
-   * @returns A list of block types matching the query.
+   * @returns A list of block states matching the query.
    */
-  blockTypesMatching(query: string): string[] {
-    return [
+  blockTypesMatching(query: string): Blockly.serialization.blocks.State[] {
```

Comment:
Because this is public, changing the return type of this method is actually a breaking change. Could you create another method `blockStatesMatching` that has the current behavior here? And then modify this to only return the block /types/ not the full state. You'll need to update other places to use your new `blockStatesMatching` method.